### PR TITLE
Upload linux-x86_64 artifacts on try branch.

### PR DESCRIPTION
**Pull request by tomprince**
This is an experiment to explore generating artifacts for testing with cargobmomb. See brson/cargobomb#64.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -153,15 +154,15 @@ before_script:
       df -h;
       du . | sort -nr | head -n100
   - >
-      if [ "$ALLOW_PR" = "" ] && [ "$TRAVIS_BRANCH" != "auto" ]; then
-          export RUN_SCRIPT="echo 'skipping, not a full build'";
-      else
+      if [[ "$ALLOW_PR" = "1" || "$TRAVIS_BRANCH" == "auto" || ( "$ALLOW_TRY" = "1" && "$TRAVIS_BRANCH" = "try" ]; then
```
Comment (Mark-Simulacrum) [inline .travis.yml]: The `(` appears to be unclosed.

**Review by Mark-Simulacrum — CHANGES_REQUESTED** [review]

**Review by Mark-Simulacrum — APPROVED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -153,15 +153,29 @@ before_script:
...
-      if [[ "$ALLOW_PR" = "1" || "$TRAVIS_BRANCH" == "auto" || ( "$ALLOW_TRY" = "1" && "$TRAVIS_BRANCH" = "try" ) ]]; then
+      # If we are building a pull request, do the build if $ALLOW_PR == 1
+      if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then
+          if [[ "$ALLOW_PR" == "1" ]]; then
+              SKIP_BUILD=false
+          else
+              SKIP_BUILD=true
+          fi
+      # Otherwise, do the build if we are on the auto branch, or the try branch and $ALLOW_TRY == 1
+      elif [[ "$TRAVIS_BRANCH" == "auto" || ( "$ALLOW_TRY" == "1" || "$TRAVIS_BRANCH" == "try" ) ]]; then
```
Comment (Mark-Simulacrum) [inline .travis.yml]: Allow try && try branch

**Review by Mark-Simulacrum — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (Mark-Simulacrum) [inline .travis.yml]: Actually, no, this is correct. Never mind.

**Review by Mark-Simulacrum — APPROVED** [review]

File: .travis.yml (hunk 2 above)
Comment (tomprince) [inline .travis.yml]: Your first comment was correct.

**Comment by tomprince** [comment]
Can somebody trigger another try run of this before landing, to check that the build starts? Once the `IMAGE=dist-x86_64-linux DEPLOY=1 ALLOW_TRY=1` has started running, it can be canceled.

**Comment by tomprince** [comment]
The artifacts got successfully uploaded to `s3://rust-lang-ci/rustc-builds-try/69feff5004e7a1844708151ebd9a57d576dadfc7/` from the try build.

**Comment by Mark-Simulacrum** [comment]
I don't know of homu is supposed to print something, but try succeeded. I guess that makes this successful -- r? @alexcrichton for final sign off though.

**Comment by tomprince** [comment]
Here are the builds:
- [PR](https://travis-ci.org/rust-lang/rust/builds/239236626) (which only has `IMAGE=x86_64-gnu-llvm-3.7 ALLOW_PR=1 RUST_BACKTRACE=1` running)
- [`try`](https://travis-ci.org/rust-lang/rust/builds/239292712) (which only has `IMAGE=dist-x86_64-linux DEPLOY=1 ALLOW_TRY=1` running and has artifacts at s3://rust-lang-ci/rustc-builds-try/d33ebd0b1c28aeb1823df5314c578586feaa6bd6/

As this merges,
- The `auto` build should build everything.
- The `master`build should build nothing.

**Comment by Mark-Simulacrum** [comment]
Actually, I believe @alexcrichton gave approval over IRC yesterday.

@bors r+

**Pull request merged**